synopsys-task/src/main.ts

```diff
@@ -72,19 +85,72 @@ export async function run() {
   console.log("Synopsys Task workflow execution completed");
 }
 
-export function logBridgeExitCodes(message: string): string {
-  const exitCode = message.trim().slice(-1);
+export function getExitMessage(message: string, exitCode: string): string {
   return constants.EXIT_CODE_MAP.has(exitCode)
-    ? "Exit Code: " + exitCode + " " + constants.EXIT_CODE_MAP.get(exitCode)
-    : message;
+    ? "Exit Code: " + exitCode + " - " + constants.EXIT_CODE_MAP.get(exitCode)
+    : "Exit Code: " +
+        ErrorCode.UNDEFINED_ERROR_FROM_EXTENSION.toString() +
+        " - " +
+        "Undefined error from extension: " +
+        message;
 }
 
-run().catch((error) => {
-  if (error.message != undefined) {
-    taskLib.error(error.message);
+export function getStatusFromError(errorObject: Error): string {
+  return errorObject.message.trim().split(" ").pop() || "";
+}
+
+function markBuildStatusIfIssuesArePresent(
+  status: string,
+  taskResult: TaskResult,
+  errorMessage: string
+) {
+  const exitMessage = getExitMessage(errorMessage, status);
+
+  if (status == ErrorCode.BRIDGE_BREAK_ENABLED.toString()) {
+    console.log(errorMessage);
+    if (taskResult === TaskResult.Succeeded) {
+      console.log(exitMessage);
+    }
+    console.log(
+      `Marking the build ${TaskResult[taskResult]} as configured in the task`
+    );
+    taskLib.setResult(taskResult, exitMessage);
+  } else {
+    taskLib.error(errorMessage);
+    console.log(
+      `Marking build status ${TaskResult[taskResult]} is ignored since exit code is: ${status}`
```

Comment:
I believe, this log message can be defined as constant

Reply:
Updated as constant